### ISSUE TITLE
WooCommerce Notices - Product Single Template - Elementor Pro

### ISSUE THREAD

**Issue by gamertan** [elementor/elementor#8124]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
When designing a Single Product template in Elementor Pro, we are unable to specify where WooCommerce notices echo out to, how they look, or when they appear. We end up with all of the WooCommerce notices appearing above the Elementor template, and subsequently breaking our WooCommerce templates. Specifically, it's incredibly annoying to see "product added to cart" above (underneath a transparent) navigation. 

The WooCommerce notices echo out into the hook `woocommerce_before_single_product`. If we had an element that would echo out the code required into our preferred location using elementor styling, we'd be better able to control how it displays. 

**Describe the solution you'd like**
A simple element that would echo out the code used in the WooCommerce `content-single-product.php` template would do the trick. Being able to style the element with Elementor would be ideal. This would allow us to place the notifications wherever we find appropriate based on our templates.

```
/**
 * Hook: woocommerce_before_single_product.
 *
 * @hooked wc_print_notices - 10
 */
do_action( 'woocommerce_before_single_product' );
```
https://github.com/woocommerce/woocommerce/blob/3.5.0/templates/content-single-product.php

**Describe alternatives you've considered**
I don't think there are any alternatives that would complete such a task with such simplicity. Sure, you would engineer the notifications into another element, or add them automatically to some section, but it wouldn't uphold the "customizability" that Elementor promotes so highly.

**Additional context**
If an element like "WooCommerce Pages" exists, I would expect that other WooCommerce hooks/shortcodes/features would be extended just the same. If I have to dive into a php template, I'm going to assume it should be completed by Elementor and one of its elements.

**Comment by edmundcwm** [comment]
Facing the same issue as well. When using Elementor to design Woocommerce pages, does it really break the core WC structure, rendering all hooks useless?

**Comment by tekatone** [comment]
This is driving me nuts. Just registered to comment on this issue. Can we have any update if this issue been considered as a bug? Or any future plans to fix it?

**Comment by formulaclick** [comment]
Same problem here. Any solution?

**Comment by Akaaal** [comment]
Same here, really annoying...

**Comment by mswiaczkowski** [comment]
I don't know if this is elegant or not, but this is how I handle it and it works.

Add this to your child theme's functions.php file:

`remove_action( 'woocommerce_before_single_product', 'woocommerce_output_all_notices', 10 );
`

Then place a Shortcode Widget anywhere you would like to put the notices and paste there this:

`[shop_messages]`

**Comment by gamertan** [comment]
> Add this to your child theme's functions.php file:
> 
> `remove_action( 'woocommerce_before_single_product', 'woocommerce_output_all_notices', 10 ); `
> 
> Then place a Shortcode Widget anywhere you would like to display notices and paste there this:
> 
> `[shop_messages]`

Thanks for the alternative solution for the time being.

Though, this is basically what I'm requesting be added to Elementor as a Widget. I'm just looking for a native solution to something that should be considered key functionality. Considering the audience for Elementor, it should encapsulate all "core" functionality in it's drag and drop page builder interface / options.

**Comment by IlPizza79** [comment]
REALLY thanks for the workaround!
But let's wait for an "elementor widget" solution...

**Comment by zerfl** [comment]
> REALLY thanks for the workaround!
> But let's wait for an "elementor widget" solution...

People have waited for over 2 years. I don't see this happening at all.

**Comment by avivu** [comment]
Hello,
This issue has been resolved in Elementor Pro v3.6.6

Feel free to update

Thanks!
